tap_zendesk/streams.py

```diff
@@ -182,10 +182,13 @@ def sync(self, state):
             # need to check total response size before iterating
             # See: https://develop.zendesk.com/hc/en-us/articles/360022563994--BREAKING-New-Search-API-Result-Limits
             if users.count > 1000:
-                search_window_size = search_window_size // 2
-                end = start + datetime.timedelta(seconds=search_window_size)
-                LOGGER.info("users - Detected Search API response size too large. Cutting search window in half to %s seconds.", search_window_size)
-                continue
+                if search_window_size > 1:
+                    search_window_size = search_window_size // 2
+                    end = start + datetime.timedelta(seconds=search_window_size)
+                    LOGGER.info("users - Detected Search API response size too large. Cutting search window in half to %s seconds.", search_window_size)
+                    continue
+
+                raise Exception("users - Unable to get all users within minimum window of a single second ({}), found {} users within this timestamp. Zendesk can only provide a maximum of 1000 users per request. See: https://develop.zendesk.com/hc/en-us/articles/360022563994--BREAKING-New-Search-API-Result-Limits".format(parsed_start, users.count))
```

Comment:
This Exception will conditionally execute based on the results of the `if search_window_size > 1:`, can we put it in an else block to call that out?

Reply:
Pylint complains about that with some kind of "else after continue" :(